### ISSUE TITLE
두근득근 일지 CRUD의 C 만들기.

### ISSUE THREAD

**Issue by JungDaeSuh** [dnwlsrla40/heart-muscle#11]
- [ ] 두근득근 일지 CRUD의 C 만들기.
- [ ] "좋아요" 버튼.
- [ ] "조회수" 추가.

C: 이름, 제목, 내용
R: 번호, 작성시간, 작성자, 제목, 내용, 추천(좋아요).
U: 추천(좋아요). 

조회수 기능은 이번 주에 만들기 보다는 수업을 더 듣고 만드는 것이 나아보임. 

클라이언트
`<!doctype html>
<html lang="en">
<head>
    
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.0/dist/css/bootstrap.min.css"
          integrity="sha384-B0vP5xmATw1+K9KRQjQERJvTumQW0nPEzvF6L/Z6nronJ3oUOFUFpCjEUQouq2+l" crossorigin="anonymous">

    
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.0/dist/js/bootstrap.bundle.min.js"
            integrity="sha384-Piv4xVNRyMGpqkS2by6br4gNJ7DXjqk09RmUpJ8jgGtD7zP9yug3goQfGII0yAns"
            crossorigin="anonymous"></script>

    <script defer src="https://use.fontawesome.com/releases/v5.3.1/js/all.js"></script>

    <title>득근일지</title>
    <script>
        $(function () {
            showLog();
        });

        function saveLog() {
            let name = $('#name').val()
            let title = $('#title').val()
            let content = $('#content').val()

            $.ajax({
                type: "POST",
                url: "/log",
                data: {name_give: name, title_give: title, content_give: content},
                success: function (response) {
                    alert(response["msg"]);
                    window.location.reload();
                }
            })
        }

        function showLog() {
            $.ajax({
                type: "GET",
                url: "/log",
                data: {},
                success: function (response) {
                    let logs = response['all_logs'];
                    for (let i = 0; i < logs.length; i++){
                        let date = logs[i]['date']
                        let name = logs[i]['name']
                        let title = logs[i]['title']
                        let content = logs[i]['content']
                        let like = logs[i]['like']

                        let temp_html = `<tr>
                                            <th scope="row">${i+1}</th>
                                            <td>${date}</td>
                                            <td>${name}</td>
                                            <td>${title}</td>
                                            <td>${content}</td>
                                            <td>
                                                <a href="#" target="_blank" class="star-name title is-4">좋아요: ${like}</a>
                                                <a href="#" onclick="likeStar('${name}')" class="card-footer-item has-text-info"><span class="icon"><i class="fas fa-thumbs-up"></i></span></a>
                                            </td>
                                            <td></td>
                                        </tr>`
                        $('#log_table').append(temp_html)
                    }
                }
            })
        }

        function likeStar(name) {
            $.ajax({
                type: 'POST',
                url: '/api/like',
                data: {name_give:name},
                success: function (response) {
                    alert(response['msg']);
                    window.location.reload()
                }
            });
        }

    </script>

    <style>
        #inputGroup-sizing-default {
            padding-right: 20px;
        }
    </style>
</head>
<body>
    <div class="jumbotron jumbotron-fluid">
        <div class="container">
            <h1 class="display-4">두근득근 일지</h1>
            <p class="lead">오늘은 얼마나 득근을 했을까?</p>
            <div class="input-group mb-3">
                <div class="input-group-prepend">
                    <span class="input-group-text">이름</span>
                </div>
                <input type="text" class="form-control" id="name" placeholder="이름을 입력하세요" aria-label="Username"
                       aria-describedby="basic-addon1">
            </div>
            <div class="input-group mb-3">
                <div class="input-group-prepend">
                    <span class="input-group-text">제목</span>
                </div>
                <input type="text" class="form-control" id="title" placeholder="제목을 입력하세요" aria-label="Username"
                       aria-describedby="basic-addon1">
            </div>
            <div class="input-group mb-3">
                <div class="input-group-prepend">
                    <span class="input-group-text">내용</span>
                </div>
                <textarea class="form-control" id="content" placeholder="내용을 입력하세요" aria-label="With textarea"></textarea>
            </div>
            <button onclick="saveLog()" type="button" class="btn btn-secondary">두근득근 일지 작성</button>
        </div>
    </div>
    <table class="table">
        <thead>
        <tr>
            <th scope="col">번호</th>
            <th scope="col">작성시간</th>
            <th scope="col">작성자</th>
            <th scope="col">제목</th>
            <th scope="col">내용</th>
            <th scope="col">추천</th>
            <th scope="col">조회수</th>
        </tr>
        </thead>
        <tbody id="log_table">

        </tbody>

    </table>
</body>
</html>`

서버
`from flask import Flask, render_template, jsonify, request
app = Flask(__name__)

from pymongo import MongoClient
client = MongoClient('localhost', 27017)
db = client.heart_muscle

from datetime import datetime

## HTML을 주는 부분
@app.route('/')
def home():
    return render_template('heart_log.html')

## API 역할을 하는 부분
@app.route('/log', methods=['POST'])
def write_review():
    name_receive = request.form['name_give']
    title_receive = request.form['title_give']
    content_receive = request.form['content_give']
    like_receive = 0

    today = datetime.now()
    mytime = today.strftime('%Y년 %m월 %d일 %H시 %M분 %S초')

    date = f'{mytime}'

    doc = {
        'date': date,
        'name': name_receive,
        'title': title_receive,
        'content': content_receive,
        'like': like_receive
    }

    db.heart_log.insert_one(doc)

    return jsonify({'msg': '저장 완료!'})

@app.route('/api/like', methods=['POST'])
def like_star():
    name_receive = request.form['name_give']

    target_log = db.heart_log.find_one({'name': name_receive})
    current_like = target_log['like']

    new_like = current_like + 1

    db.heart_log.update_one({'name': name_receive}, {'$set': {'like': new_like}})

    return jsonify({'msg': '좋아요 완료!'})

@app.route('/log', methods=['GET'])
def read_reviews():
    logs = list(db.heart_log.find({}, {'_id': False}))
    return jsonify({'all_logs': logs})

if __name__ == '__main__':
    app.run('0.0.0.0', port=5000, debug=True)`

**Comment by JungDaeSuh** [comment]
1차 프로젝트 마무리하여, issue도 close!
